<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Frame</class>
 <widget class="QFrame" name="Frame">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>808</width>
    <height>695</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Frame</string>
  </property>
  <property name="frameShape">
   <enum>QFrame::StyledPanel</enum>
  </property>
  <property name="frameShadow">
   <enum>QFrame::Raised</enum>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QWidget" name="widget" native="true">
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <layout class="QGridLayout" name="gridLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="x_pos_label">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>X position</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLineEdit" name="x_pos_edit">
            <property name="minimumSize">
             <size>
              <width>80</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>80</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="toolTip">
             <string>Position of the beam in X, in pixels</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLabel" name="y_pos_label">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Y position</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="y_pos_edit">
            <property name="minimumSize">
             <size>
              <width>80</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>80</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="toolTip">
             <string>Position of the beam in Y, in pixels.</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QCheckBox" name="use_beam_finder_checkbox">
          <property name="toolTip">
           <string>Select to let the reduction software find the beam center.</string>
          </property>
          <property name="text">
           <string>Use beam finder</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>Beam Finder</string>
          </property>
          <widget class="QWidget" name="formLayoutWidget">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>20</y>
             <width>703</width>
             <height>118</height>
            </rect>
           </property>
           <layout class="QFormLayout" name="formLayout">
            <property name="sizeConstraint">
             <enum>QLayout::SetFixedSize</enum>
            </property>
            <property name="fieldGrowthPolicy">
             <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
            </property>
            <property name="margin">
             <number>5</number>
            </property>
            <item row="0" column="1">
             <layout class="QHBoxLayout" name="horizontalLayout"/>
            </item>
            <item row="1" column="0">
             <layout class="QHBoxLayout" name="horizontalLayout_4">
              <item>
               <widget class="QLabel" name="data_file_label">
                <property name="text">
                 <string>Data file:                       </string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="beam_data_file_edit">
                <property name="minimumSize">
                 <size>
                  <width>300</width>
                  <height>0</height>
                 </size>
                </property>
                <property name="toolTip">
                 <string>Enter the file path of a direct beam data file.</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="data_file_browse_button">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="text">
                 <string>Browse...</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="2" column="0">
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <item>
               <widget class="QRadioButton" name="direct_beam">
                <property name="toolTip">
                 <string>Fit the direct beam to obtain the beam center.</string>
                </property>
                <property name="text">
                 <string>Fit direct beam</string>
                </property>
                <property name="checked">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QRadioButton" name="scattering_data">
                <property name="toolTip">
                 <string>Fit only the scattering profile to obtain the beam center.</string>
                </property>
                <property name="text">
                 <string>Fit scattering data</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="3" column="0">
             <layout class="QHBoxLayout" name="horizontalLayout_3">
              <item>
               <widget class="QLabel" name="beam_radius_label">
                <property name="text">
                 <string>Beam radius [pixels]</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QLineEdit" name="beam_radius_edit">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="minimumSize">
                 <size>
                  <width>97</width>
                  <height>0</height>
                 </size>
                </property>
                <property name="toolTip">
                 <string>Enter the radius of the beam in pixels.</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>use_beam_finder_checkbox</sender>
   <signal>clicked()</signal>
   <receiver>Frame</receiver>
   <slot>use_beam_finder_changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>108</x>
     <y>105</y>
    </hint>
    <hint type="destinationlabel">
     <x>370</x>
     <y>5</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>scattering_data</sender>
   <signal>clicked()</signal>
   <receiver>Frame</receiver>
   <slot>fit_scattering_data()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>255</x>
     <y>198</y>
    </hint>
    <hint type="destinationlabel">
     <x>580</x>
     <y>5</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>direct_beam</sender>
   <signal>clicked()</signal>
   <receiver>Frame</receiver>
   <slot>fit_direct_beam()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>97</x>
     <y>196</y>
    </hint>
    <hint type="destinationlabel">
     <x>60</x>
     <y>4</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>use_beam_finder_changed()</slot>
  <slot>fit_scattering_data()</slot>
  <slot>fit_direct_beam()</slot>
 </slots>
</ui>
